source/common/common/envoy_defaults.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include "source/common/common/manifest.pb.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/singleton/threadsafe_singleton.h"
+
+// These macros return the value associated with `field_name` in the protobuf `message` if it
+// exists. Otherwise, they return the value associated with `field_name` in the defaults profile if
+// it exists. Otherwise, `default_value` is returned.
+
+#define PROTOBUF_GET_NUMBER_OR_PROFILE_DEFAULT(message, config, field_name, default_value)         \
```

Comment:
nit: I think all uses of the new macro have `DefaultsProfile::ConfigContext(config)` as a second parameter, so we do not actually need it to be a separate parameter, and just use the `DefaultsProfile::ConfigContext(message)` inside the macro.

We can clean it up later on.